### PR TITLE
refactor(animate): add the new operator to injected $q

### PR DESCRIPTION
**Pull request by pocesar**
When replacing $q with Bluebird, it complains `this is undefined`. Missing the new keyword

### CODE REVIEW

**Comment by pocesar** [comment]
I signed it!

File: src/ng/animate.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ var $$CoreAnimateRunnerProvider = function() {
       pause: noop,
       complete: noop,
       then: function(pass, fail) {
-        return $q(function(resolve) {
+        return new Q(function(resolve) {
```
Comment (Narretz) [inline src/ng/animate.js]: It's not `Q`, it's `$q`.

Comment (pocesar) [inline src/ng/animate.js]: jshint complains "A constructor name should start with an uppercase letter"

**Comment by lgalfaso** [comment]
I think this is a bad idea. If someone wants to replace `$q` with Bluebird, then when the code that intercepts and replaces `$q` should take care of this

**Comment by pocesar** [comment]
Bluebird doesn't allow `Bluebird(function(resolve, reject) {})`, needs to use new on it; returning from the constructor is bad practice anyway

**Comment by lgalfaso** [comment]
Let me reiterate what I just wrote

> If someone wants to replace $q with Bluebird, then **the code that intercepts and replaces $q should take care of this**

Not Angular, not Bluebird, but the code that replaces `$q`

**Comment by lgalfaso** [comment]
No feedback in a month, closing

**Pull request closed without merging**